middleware/kubernetes/README.md

```diff
@@ -1,88 +1,126 @@
-        # that match this label selector. The label
-        # selector syntax is described in the kubernetes
-        # API documentation: http://kubernetes.io/docs/user-guide/labels/
-        # Example selector below only exposes objects tagged as
-        # "application=nginx" in the staging or qa environments.
-        #labels environment in (staging, qa),application=nginx
-		
-        # The mode of responding to pod A record requests. 
-        # e.g 1-2-3-4.ns.pod.zone.  This option is provided to allow use of
-        # SSL certs when connecting directly to pods.
-        # Valid values: disabled, verified, insecure
-        #  disabled: default. ignore pod requests, always returning NXDOMAIN
-        #  insecure: Always return an A record with IP from request (without 
-        #            checking k8s).  This option is is vulnerable to abuse if
-        #            used maliciously in conjuction with wildcard SSL certs.
+This is an example kubernetes middle configuration block, with all options described:
+
+```
+# kubernetes <zone> [<zone>] ...
+#
+# Use kubernetes middleware for domain "coredns.local"
+# Reverse domain zones can be defined here (e.g. 0.0.10.in-addr.arpa), 
+# or instead with the "cidrs" option.
+#
+kubernetes coredns.local {
+	
+	# resyncperiod <period>
+	#
+	# Kubernetes data API resync period. Default is 5m
+	# Example values: 60s, 5m, 1h
+	#
+	resyncperiod 5m
+	
+	# endpoint <url>
+	#
+	# Use url for k8s API endpoint. Default is https://localhost:8080
+	#
+	endpoint https://localhost:8080
+
+	# tls <cert-filename> <key-filename> <cacert-filename>
+	#
+	# The tls cert, key and the CA cert filenames
+	#
+	tls cert key cacert
+	
+	# namespaces <namespace> [<namespace>] ...
+	#
+	# Only expose the k8s namespaces listed.  If this option is omitted
+	# all namespaces are exposed
+	#
+	namespaces demo
+	
+	# lables <expression> [,<expression>] ...
+	#
+	# Only expose the records for kubernetes objects
+	# that match this label selector. The label
+	# selector syntax is described in the kubernetes
+	# API documentation: http://kubernetes.io/docs/user-guide/labels/
+	# Example selector below only exposes objects tagged as
+	# "application=nginx" in the staging or qa environments.
+	#
+	labels environment in (staging, qa),application=nginx
+	
+	# pods <disabled|insecure|verified>
+	#
+	# Set the mode of responding to pod A record requests. 
+	# e.g 1-2-3-4.ns.pod.zone.  This option is provided to allow use of
+	# SSL certs when connecting directly to pods.
+	# Valid values: disabled, verified, insecure
+	#  disabled: Do not process pod requests, always returning NXDOMAIN
+	#  insecure: Always return an A record with IP from request (without 
+	#            checking k8s).  This option is is vulnerable to abuse if
+	#            used maliciously in conjuction with wildcard SSL certs.
+	#  verified: Return an A record if there exists a pod in same 
+	#            namespace with matching IP.
```

Comment:
Add: Requires substantially more memory than the `insecure` mode, as it will keep a watch on all pods.